File: app/src/main/java/co/wm21/https/adapters/category/CategoryView.java
package co.wm21.https.adapters.category;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class CategoryView {
    @SerializedName("categoryName")
    @Expose
    private String categoryName;
    @SerializedName("orderingID")
    @Expose
    private String orderingID;
    @SerializedName("catID")
    @Expose
    private String catID;
    @SerializedName("categoryImageUrl")
    @Expose
    private String categoryImageUrl;

    public CategoryView(String categoryName, String categoryImageUrl) {
        this.categoryName = categoryName;
        this.categoryImageUrl = categoryImageUrl;
    }
    public CategoryView(String categoryName, String categoryImageUrl,String catID) {
        this.categoryName = categoryName;
        this.categoryImageUrl = categoryImageUrl;
        this.catID = catID;
    }

    public String getCatID() {
        return catID;
    }

    public void setCatID(String catID) {
        this.catID = catID;
    }

    public String getCategoryName() {
        return categoryName;
    }

    public void setCategoryName(String categoryName) {
        this.categoryName = categoryName;
    }

    public String getCategoryImageUrl() {
        return categoryImageUrl;
    }

    public void setCategoryImageUrl(String categoryImageUrl) {
        this.categoryImageUrl = categoryImageUrl;
    }

    public String getOrderingID() {
        return orderingID;
    }

    public void setOrderingID(String orderingID) {
        this.orderingID = orderingID;
    }
}

File: app/src/main/java/co/wm21/https/FHelper/networks/Models/home/CategoryView.kt
package co.wm21.https.FHelper.networks.Models.home

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class CategoryView {
    @JvmField
    @SerializedName("categoryName")
    @Expose
    var categoryName: String

    @SerializedName("orderingID")
    @Expose
    var orderingID: String? = null

    @JvmField
    @SerializedName("catID")
    @Expose
    var catID: String? = null

    @JvmField
    @SerializedName("categoryImageUrl")
    @Expose
    var categoryImageUrl: String

    constructor(categoryName: String, categoryImageUrl: String) {
        this.categoryName = categoryName
        this.categoryImageUrl = categoryImageUrl
    }

    constructor(categoryName: String, categoryImageUrl: String, catID: String?) {
        this.categoryName = categoryName
        this.categoryImageUrl = categoryImageUrl
        this.catID = catID
    }
}
